src/query/compiler.js

```diff
@@ -46,6 +46,11 @@ assign(QueryCompiler.prototype, {
     if (method === 'select' && this.single.as) {
       defaults.as = this.single.as;
     }
+
+    if(this.client && this.client.prepBindings) {
+      defaults.bindings = this.client.prepBindings(defaults.bindings);
+    }
+
```

Comment:
I'm not sure if this is correct place to put prepare bindings... e.g. in that case it will be ran multiple times for `queryBuilder.toString()` or `queryBuilder.toQuery()` methods (https://github.com/tgriesser/knex/blob/master/src/interface.js#L7).

 
Looks like there is already code in dialect side, which tries to do it in MSSQL https://github.com/tgriesser/knex/blob/master/src/dialects/mssql/index.js#L103 and oracle https://github.com/tgriesser/knex/blob/master/src/dialects/oracle/index.js#L129
but for some reason postgres doesn't have it in place https://github.com/tgriesser/knex/blob/master/src/dialects/postgres/index.js#L132 it wasn't there either for SQLite and I think that oracles stream version of query didn't call that either...

Probably it would be safest to call it somewhere in common code, like maybe here, but in that case we need to remove calls to it from other places. And all the places where `toSQL` is used should be checked that preparing bindings doesn't get called multiple times as a side effect... 

It could make sense though that when we get bindings we prepare them right away and then bindings everywhere would be already prepared... only bad thing in that would be that if client is given to e.g. `knex.raw` later on, bindings will not be updated correctly for that client.